packages/purpose-process/src/model/domain/errors.ts

```diff
@@ -347,3 +348,13 @@ export function puroposeDelegationNotFound(
     title: "Purpose delegation not found",
   });
 }
+
+export function purposeCannotBeUpdated(
+  purposeId: PurposeId
+): ApiError<ErrorCodes> {
+  return new ApiError({
+    detail: `Purpose ${purposeId} cannot be updated`,
```

Comment:
If it's only the case for Archived purpose we could be more specific in the error
```suggestion
    detail: `Archived purpose ${purposeId} cannot be updated`,
```